packages/plugin-preview/src/generate-entry.ts

```diff
@@ -4,11 +4,11 @@ import { staticPath, virtualDir } from './constant';
 import type { DemoInfo } from './types';
 import { toValidVarName } from './utils';
 
-// TODO: Support custom entry template files
 export function generateEntry(
   demos: DemoInfo,
   framework: 'react' | 'solid',
   position: 'follow' | 'fixed',
+  customEntry?: (entryCssPath: string, demoPath: string) => string,
```

Comment:
```suggestion
  customEntry?: ({ entryCssPath: string, demoPath: string }) => string,
```

NIT: A more extensible change would be better with object types.